pkg/api/kptfile/v1alpha2/types.go

```diff
@@ -40,15 +40,14 @@ var TypeMeta = yaml.ResourceMeta{
 type KptFile struct {
 	yaml.ResourceMeta `yaml:",inline"`
 
+	Upstream *Upstream `yaml:"upstream,omitempty"`
```

Comment:
Will Kptfile have both Upstream and UpsteamLock ? Why can't we just go down the current(master) route and just have Upstream ?

Reply:
We need both information about the upstream used for the current version of the local package, which used to be the `upstream` section of the package Kptfile, but which is now included in the `upstreamLock` section. This is information that is always automatically updated by kpt.
We also need to allow the user to declaratively specify the desired upstream fro the package. This information used to be in the `subpackages` section of the parent Kptfile, but we have now moved it into the package itself. The update process consists of two steps, first update the "desired" upstream by changing the information in the `upstream` section (either manually or through kpt porcelain commands), and then ask kpt to perform the update by pulling down the desired version and performing the merge.

Reply:
So a Kptfile contains both Upstream and UpstreamLock section at the same time ? How does a Kptfile look like? I think that is confusing. How would a user know what to edit ? 

```
The update process consists of two steps, first update the "desired" upstream by changing the information in the upstream section (either manually or through kpt porcelain commands), and then ask kpt to perform the update by pulling down the desired version and performing the merge.
```

This is the case even with current form of kpt. I think having just one Upstream section is simple and intuitive. Users can directly edit Upstream section. Kpt can see that section, perform update/merge, update commit hash.

Reply:
I think keeping information that is managed by kpt separate from the information we expect users to update makes it easier to understand.